### PR TITLE
Configure Infrastructure logger

### PR DESCRIPTION
**Pull request by piecyk**
**What kind of change does this PR introduce?**

Would be possible to configure infrastructure logger. For example overrides colors for some specific plugin. 
Related to #11572

**Did you add tests for your changes?**

Yes.

**Does this PR introduce a breaking change?**

Yes.  If someone overrides the webpack console impl via options.infrastructureLogging, it changed from `console` to `getConsole` getter.

**What needs to be documented once your changes are merged?**

https://webpack.js.org/api/logging/

### CODE REVIEW

File: lib/Compiler.js [hunk 1]
```diff
@@ -154,8 +154,13 @@ class Compiler {
 			/** @type {SyncHook<[]>} */
 			watchClose: new SyncHook([]),
 
-			/** @type {SyncBailHook<[string, string, any[]], true>} */
-			infrastructureLog: new SyncBailHook(["origin", "type", "args"]),
+			/** @type {SyncBailHook<[string, object | undefined, string, any[]], true>} */
+			infrastructureLog: new SyncBailHook([
+				"origin",
+				"options",
```
Comment (evilebottnawi) [inline lib/Compiler.js:160]: Breaking change

Comment (piecyk) [reply lib/Compiler.js:160]: What we can do with this? 🤔

Comment (evilebottnawi) [reply lib/Compiler.js:160]: Adding this as last arg :smile:

Comment (piecyk) [reply lib/Compiler.js]: Yeah 😅

**Comment by piecyk** [comment]
Let's rethink this approach, closing for now.

**Pull request closed without merging**